src/platform/unix/mod.rs

```diff
@@ -435,61 +443,76 @@ impl OsIpcReceiverSet {
         let fnv = BuildHasherDefault::<FnvHasher>::default();
         Ok(OsIpcReceiverSet {
             incrementor: Incrementor::new(),
-            pollfds: Vec::new(),
-            fdids: HashMap::with_hasher(fnv)
+            poll: match Poll::new() {
+                Ok(poll) => poll,
+                Err(e) => { return Err(e.into()); }
```

Comment:
I somehow missed this before (sorry): but if you implemented the `From` trait instead of the `Into` one -- like the other back-ends do -- you could just use `try!()` here I believe?...